### DISCUSSION TITLE
Use as an IDE Plugin

### DISCUSSION THREAD

**Discussion by AntumDeluge** [geany/geany#3896] [General]
I'm wondering if it is, or would be, possible to use Geany as an IDE plugin. I use [Eclipse](https://eclipse.org/) for one of the projects I am involved with, but it doesn't have any editors that I like for JavaScript/TypeScript. I _love_ Geany for doing just that. But, before I make too much effort in attempting to use it as such, I want to know if it is currently or theoretically possible for Geany's editor to be used as an Eclipse plugin.

**Comment by elextr** [comment]
Eclipse plugins are Java, Geany is C, so never the twain shall meet.

But IIRC Eclipse allows external editors to be used for specified filetypes, maybe search eclipse docs, or google it.

**Comment by AntumDeluge** [comment]
> Eclipse plugins are Java, Geany is C, so never the twain shall meet.

Eclipse supports cross-language plugin development with C/C++ using the [Java Native Interface](https://en.wikipedia.org/wiki/Java_Native_Interface).

> But IIRC Eclipse allows external editors to be used for specified filetypes, maybe search eclipse docs, or google it.

Yes, I can set file type associations to external editors in Eclipse. I just don't like having two editors open.

Theoretically, if Geany or a Geany library, had a visible function for accessing the editor interface it could be used as plugin. Sorry that I didn't research it more before asking. I was just curious if Geany already had built-in methods for exposing its editor to external applications.
